libs/libc/sched/sched_dumpstack.c

```diff
@@ -55,6 +55,26 @@ void sched_dumpstack(pid_t tid)
 {
   int size = DUMP_DEPTH;
   int skip;
+  FAR struct tcb_s *tcb = NULL;
+  FAR mutex_t *mutex = NULL;
+
+  /* If task is in TSTATE_WAIT_SEM, it may be waiting for a mutex,
+   * then print mutex holder to help debugging.
+   * However, if it is not a mutex, an invalid holder is printed.
+   * I think printing invalid value is acceptable since this is
+   * for debugging only.
+   */
+
+  if (tid >= 0)
```

Comment:
why not move to ps and crash dump?

Reply:
cmd_ps seems a good place to go.
What do you mean by crash dump? board_crashdump?

Reply:
e.g. arm_dumpstate